autogpt_platform/backend/backend/executor/manager.py

```diff
@@ -454,6 +465,51 @@ def validate_exec(
     return data, node_block.name
 
 
+GRAPH_EXECUTION_EXCHANGE = Exchange(
+    name="graph_execution",
+    type=ExchangeType.DIRECT,
+    durable=True,
+    auto_delete=False,
+)
+GRAPH_EXECUTION_QUEUE_NAME = "graph_execution_queue"
+GRAPH_EXECUTION_ROUTING_KEY = "graph_execution.run"
+
+GRAPH_EXECUTION_CANCEL_EXCHANGE = Exchange(
+    name="graph_execution_cancel",
+    type=ExchangeType.FANOUT,
+    durable=True,
+    auto_delete=True,
+)
```

Comment:
Heavily consider topic or header rather than direct. Changing exchange types later sucks

Also maybe fanout is good if you want a broadcast notification style. Not sure goal with that

Reply:
I think for this use case we should avoid complicated filtering of the routing and should stick to having a queue of entries that is load balanced by executors. 

> Changing exchange types later sucks
Why is that so? Why can't we just introduce a new Exchange with different name & type then push ?

> fanout is good if you want a broadcast notification style. Not sure goal with that
It's obviously not the right use for execution queue, but intended for cancel execution, since all executors need to check if the execution is present on their side and cancel it accordingly.

Reply:
Sounds good just want to make sure we like what we have